### ISSUE TITLE
Do not require a minimum of 1 release branch

### ISSUE THREAD

**Issue by gempain** [semantic-release/semantic-release#1706]
## New feature motivation

When you start developing an application, you generally start by releasing alpha/beta/rc and do not have a "release branch". semantic-release doesn't allow this at the moment, as you need a minimum of 1 release branch.

## New feature description

Allow to have only pre-release branches.

**Comment by gr2m** [comment]
sounds like a reasonable request to me. I want us to take care of the `master/main` situation first, mostly to make the default release branch configurable besides the `branches` configuration, see https://github.com/semantic-release/semantic-release/issues/1581

**Comment by adrianbruntonsagecom** [comment]
I'd like to comment here in reference to #3046. Wouldn't removing the lower limit (1) also solve the linked feature request? Removing the lower limit and allowing no release branches at all would allow prereleases to be made on merge to master:
`npx semantic-release --branches '[{"name":"master","prerelease":"rc"}]'`

And full releases to be made by a manual trigger without needing to merge to a specific release branch:
`npx semantic-release --branches '[{"name":"master"}]'`

For us, this is useful because due to infrastructure requirements, package dependency versions can be inferred when building for developers (prerelease), but must be explicitly defined (meaning dependencies need to be built first) when building for release. I'm sure we "could" get around this with a specific release branch, but we are trying to minimise branches to reduce conflicts, human error and PRs etc.

I'm **assuming** an upper limit has been set for scalability reasons? Has a lower limit been set simply for "correctness"? Why not have it result in a warning, but otherwise continue with the configuration as defined if a release branch is not set?

@gr2m - given #1581 was closed two years ago, are you accepting PRs for this change, or are you looking into it?

**Comment by travi** [comment]
> I'm assuming an upper limit has been set for scalability reasons? Has a lower limit been set simply for "correctness"?

https://github.com/semantic-release/semantic-release/issues/1681 is a good discussion of why the limits are defined the way they are. Workflows that fall outside of those limits are outside of what is officially supported. Our supported workflows are opinionated and do encourage recommended workflows, so "correctness" isn't quite it, but supported our recommended is closer. 

semantic-release isn't provided as a general workflow tool, but an opinionated release tool. Beyond the opinions encouraged by semantic-release, we are a small volunteer team, so we cannot justify the maintenance that would come with supporting all variations of possible release workflows

We currently have not decided to support the linked proposal. If we were to decide to add such support, we would more formally support it than simply removing restrictions and adding more confusing configuration options that result in more support requests. 

For the time being, the workflow you are describing is not supported by semantic-release

**Comment by klopfdreh** [comment]
Hey @travi  - saw you comment here and want to follow up on that. 😺

I understand that you are afraid of what happens if you remove the upper limit and what is caused by this, but it would also a possible way to just write exactly what you mentioned here in the warning or a reference in the documentation which makes the opinion clear. 

Let me give two another example in which a similar fact exists

1. The game porting toolkit was originally mentioned to be used as a help for porting applications from windows to mac, not as a preserved way to translate games in general. It has been integrated into CrossOver and a lot of Mac users enjoy it and it's the only way to use DirectX12 through the Metal API. The opinion about that was that the applications are going to be moved to Metal API, but currently it helps to allow to run certain applications / games which were not able to run before. 

In short: Help to migrate vs general translation layer

2. Same goes for Rosetta 2. https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment 

<img width="931" height="84" alt="Image" src="https://github.com/user-attachments/assets/70b7a4bf-cb9d-494b-9dde-77a087ad0b29" />

Why to keep a subset of the Rosetta functionality when the idea was to completely remove it. It is also against the opinion that all apps are AArch64 only.

**Comment by travi** [comment]
its not a matter of fear. enabling unsupported functionality through more config options or less rigid config options results in more support requests. this is why, especially as a small, volunteer maintenance team, we have to be selective about supported features vs unsupported. we are open to being convinced of adding new supported features when there is broad enough demand from the community. we are not open to adding unpopular hidden features through additional config and just guarding with warnings. unsupported is unsupported. even in cases where unsupported features are possible through our already complex configuration options, we cannot afford to provide guidance around unsupported features, so it is even less likely that we would _add_ more unsupported capabilities
